test/cartesian/ZAxis.spec.tsx

```diff
@@ -28,6 +28,146 @@ describe('<ZAxis />', () => {
       yAxis: 1400,
     },
   ];
+
+  it('renders ZAxis with correct domain and range', () => {
+    const { container } = render(
+      <ScatterChart width={400} height={400}>
+        <XAxis dataKey="xAxis" />
+        <YAxis dataKey="yAxis" />
+        <ZAxis dataKey="zAxis" range={[100, 1000]} />
+        <Scatter data={data} />
+      </ScatterChart>,
+    );
+
+    // Instead of accessing __data__, directly check SVG elements or use testing-library utilities
+    const zAxisTicks = container.querySelectorAll('.recharts-cartesian-axis-tick');
+
+    // Check if ZAxis ticks were rendered
+    expect(zAxisTicks.length).toBeGreaterThan(0);
+  });
+
+  it('correctly applies custom ZAxis scaling', () => {
+    const { container } = render(
+      <ScatterChart width={400} height={400}>
+        <XAxis dataKey="xAxis" />
+        <YAxis dataKey="yAxis" />
+        <ZAxis dataKey="zAxis" scale="sqrt" />
+        <Scatter data={data} />
+      </ScatterChart>,
+    );
+
+    // Since we cannot directly access __data__, focus on the observable behavior
+    const symbols = container.querySelectorAll('.recharts-layer.recharts-scatter-symbol');
+    // Perform assertions based on the rendered symbols' attributes
+    expect(symbols.length).toBe(2);
+  });
+
+  it('renders with the correct number of children', () => {
+    render(
+      <Surface width={500} height={500}>
+        <ZAxis dataKey="x" name="stature" unit="cm" />
+      </Surface>,
+    );
+    const svg = document.querySelector('svg');
+    assertNotNull(svg);
+
+    expect(svg).toBeInTheDocument();
+    expect(svg.children).toHaveLength(2);
+  });
+
+  it('displays the correct tooltip content when hovering over scatter symbols', async () => {
+    const { container } = render(
+      <ScatterChart height={400} width={400}>
+        <XAxis dataKey="xAxis" type="number" />
+        <YAxis dataKey="yAxis" />
+        <ZAxis dataKey="zAxis" name="test name" range={[0, 2000]} unit="km" />
+        <CartesianGrid />
+        <Scatter data={data} name="pageData" />
+        <Tooltip />
+      </ScatterChart>,
+    );
+
+    const svg = document.querySelector('svg');
+    const gContainers = svg.querySelectorAll('.recharts-layer.recharts-scatter-symbol');
+
+    const firstShape = gContainers[0].querySelector('path');
```

Comment:
Check out the test helper methods `showTooltip` and `expectTooltipPayload`, might be easier to use.